rrrspec-client/lib/rrrspec/registry.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Registry
+
+    @@runners = Hash.new
+
+    def self.reset_runners
+      @@runners = []
+    end
+
+    def self.register_runner(factory, file_extension)
+      @@runners[file_extension] = factory
+    end
+
+
+    def self.get_runner(file_extension)
+      @@runners[file_extension]
```

Comment:
The semantics for the registry, was like this, no more factories, register the runner in in the registry and on get just return the registered object. 

Is this similar to what you were planning?